pkg/inframetadata/internal/hostmap/hostmap.go

```diff
@@ -0,0 +1,143 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package hostmap
+
+import (
+	"fmt"
+	"sync"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	conventions "go.opentelemetry.io/collector/semconv/v1.18.0"
+	"go.uber.org/multierr"
+
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata/gohai"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata/payload"
+)
+
+// HostMap maps from hostnames to host metadata payloads.
+type HostMap struct {
+	// mu is the mutex for the host map and updater.
+	mu sync.Mutex
+	// hosts map
+	hosts map[string]payload.HostMetadata
+}
+
+// New creates a new HostMap.
+func New() (*HostMap, error) {
+	return &HostMap{
+		mu:    sync.Mutex{},
```

Comment:
You can remove this line. The mutex will be initialised as its zero value.

Reply:
Done